### PR TITLE
Use pair of edge revisions instead of listing all

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
This PR replaces the slice of unapplied revisions returned from `DefineRevisions` call with an equivalent tuple of "watermark" revisions. This removes the danger of this list growing uncontrollably, and makes it easier to change the semantic of this method in #1335.

### CODE REVIEW

File: core/sequencer/sequencer_api.proto [hunk 1]
```diff
@@ -75,8 +75,18 @@ message DefineRevisionsRequest {
 
 // DefineRevisionResponse contains information about freshly defined revisions.
 message DefineRevisionsResponse {
-  // outsanding_revisions a list of all the defined revisions which are not yet applied.
-  repeated int64 outstanding_revisions = 1;
+  // map_revision is the current map revision that was fully applied. Any
```
Comment (gdbelvin) [inline core/sequencer/sequencer_api.proto]: current / highest map revision that has been fully applied

File: core/sequencer/sequencer_api.proto [hunk 2]
```diff
@@ -75,8 +75,18 @@ message DefineRevisionsRequest {
...
+  // unapplied revisions are greater than map_revision.
+  int64 map_revision = 2;
```
Comment (gdbelvin) [inline core/sequencer/sequencer_api.proto]: `highest_applied_revision`?

File: core/sequencer/sequencer_api.proto [hunk 3]
```diff
@@ -75,8 +75,18 @@ message DefineRevisionsRequest {
...
+  // old_defined is the last defined revision before this call. Any newly
+  // defined revisions are greater than old_defined.
+  int64 old_defined = 3;
+  // new_defined is the last defined revision after this call. No new revisions
+  // are defined if new_defined <= old_defined.
+  int64 new_defined = 4;
```
Comment (gdbelvin) [inline core/sequencer/sequencer_api.proto]: "new" is slightly ambiguous.  Is it the number of new defined revisions or is it the new version of defined? 

Would `highest_defined_revision` be clearer?

File: core/sequencer/server.go [hunk 4]
```diff
@@ -274,14 +274,16 @@ func (s *Server) DefineRevisions(ctx context.Context,
...
-	if len(outstanding) > 0 {
-		return &spb.DefineRevisionsResponse{OutstandingRevisions: outstanding}, nil
+	resp := &spb.DefineRevisionsResponse{
+		MapRevision: int64(latestMapRoot.Revision),
+		OldDefined:  highestRev,
+		NewDefined:  highestRev,
+	}
+	// Don't create new revisions if there are ones waiting to be applied, or the
+	// highest defined revision is lagging behind for some reason.
+	if resp.OldDefined != resp.MapRevision {
```
Comment (gdbelvin) [inline core/sequencer/server.go]: Should this logic be with respect to NewDefined?

File: core/sequencer/sequencer_api.proto [hunk 5]
```diff
@@ -75,8 +75,18 @@ message DefineRevisionsRequest {
...
 // DefineRevisionResponse contains information about freshly defined revisions.
 message DefineRevisionsResponse {
-  // outsanding_revisions a list of all the defined revisions which are not yet applied.
-  repeated int64 outstanding_revisions = 1;
+  // map_revision is the current map revision that was fully applied. Any
+  // unapplied revisions are greater than map_revision.
+  int64 map_revision = 2;
+  // old_defined is the last defined revision before this call. Any newly
+  // defined revisions are greater than old_defined.
+  int64 old_defined = 3;
```
Comment (gdbelvin) [inline core/sequencer/sequencer_api.proto]: I'm a bit unclear on why this field is needed. Should it be removed? 
If it is only to measure when new revisions have been added would a more direct "new_revisions" field be appropriate?

File: core/sequencer/sequencer_api.proto (hunk 2 above)
Comment (pavelkalinnikov) [reply core/sequencer/sequencer_api.proto]: How about `applied_to`?

Comment (pavelkalinnikov) [reply core/sequencer/sequencer_api.proto]: Renamed to `highest_applied`. I think the "revision" word is redundant.

File: core/sequencer/sequencer_api.proto (hunk 5 above)
Comment (pavelkalinnikov) [reply core/sequencer/sequencer_api.proto]: Removed.

File: core/sequencer/sequencer_api.proto (hunk 3 above)
Comment (pavelkalinnikov) [reply core/sequencer/sequencer_api.proto]: Renamed to `highest_defined`.

File: core/sequencer/server.go (hunk 4 above)
Comment (pavelkalinnikov) [reply core/sequencer/server.go]: Now irrelevant.

**Pull request merged**